### ISSUE TITLE
Open a new tab by middle clicking in empty space in the row of tabs

### ISSUE THREAD

**Issue by MazeWave** [files-community/Files#3808]
**Is your feature request related to a problem? Please describe.**
We can close tabs by middle clicking on it and open tabs by middle clicking on folders.
Can we open a new tab without using the 'add a new tab' button ?

**Describe the solution you'd like**
If you middle click in the empty space next to the 'add a new tab' button, a new tab is created

**Describe alternatives you've considered**
I've considered no alternative to this feature.

**Additional context**
![Example](https://user-images.githubusercontent.com/25009364/109882663-868a5880-7c7a-11eb-9c89-3397e8889acc.png)

**Comment by MazeWave** [comment]
Note : The best way to show this particular feature is to use Firefox browser.

**Comment by Phrygan** [comment]
Hello, I'm new to contributing to open source, is this something I can work on?

**Comment by Josh65-2201** [comment]
Yes

**Comment by WhyNot180** [comment]
Hey @Phrygan are you still working on this? Do you mind if I take a crack at it?

**Comment by Phrygan** [comment]
Go ahead, I never got the opportunity to work on it.

**Comment by yaira2** [comment]
Further research is necessary but it looks like the `PointerPressed` event isn't compatible in the title bar drag area.

**Comment by TRUINGLol** [comment]
> Further research is necessary but it looks like the `PointerPressed` event isn't compatible in the title bar drag area.
Yes it's true. Today I spent the whole day making this issue, but I still couldn’t understand why the event wasn’t triggered. I'll look into the issue tomorrow

**Comment by TRUINGLol** [comment]
> Further research is necessary but it looks like the `PointerPressed` event isn't compatible in the title bar drag area.

Yes it's true. Today I spent the whole day making this issue, but I still couldn’t understand why the event wasn’t triggered. I'll look into the issue tomorrow

**Comment by TRUINGLol** [comment]
In General, I think I understand why the PointerPressed event does not work on the title bar drag area, the fact is that this area belongs to the non-client area of ​​the window, which means it is processed in a special way. I found a class in the documentation that should help solve the click tracking problem. Here is an excerpt from the documentation

> When you add interactive content in the title bar area, you need to use the InputNonClientPointerSource class to specify areas where input is passed through to the interactive control, rather than handled by the title bar

Here are some links where I found information
https://learn.microsoft.com/en-us/windows/win32/gdi/nonclient-area

https://learn.microsoft.com/en-us/windows/apps/develop/title-bar

https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.input.inputnonclientpointersource.pointerpressed?view=windows-app-sdk-1.5#microsoft-ui-input-inputnonclientpointersource-pointerpressed

I don’t claim to be telling the truth, maybe I don’t understand how the whole project works

**Comment by om2805** [comment]
Hey I'd like to start working on this issue. Please let me know if it's available.

**Comment by yair100** [comment]
@om2805 it's available.

**Comment by Thuram2003** [comment]
Hi @yair100 , I’d like to work on this issue.

From what I understand, the main challenge is handling pointer input in the non-client title bar area using InputNonClientPointerSource.

Before I start implementing, could you confirm if this is the expected approach, or if there’s a preferred pattern already used in the project?

Thanks!

**Comment by yair100** [comment]
> From what I understand, the main challenge is handling pointer input in the non-client title bar area using InputNonClientPointerSource.

Correct

**Comment by shadyalattarr** [comment]
Hey  @Thuram2003, how's working on the issue going? I have been thinking on give it a try so I want to know how things have been doing on your end on this issue

**Comment by Thuram2003** [comment]
Hey @shadyalattarr , making good progress on it! I've dug into the codebase and have a clear picture of what needs to be done.

The main challenge is exactly what was mentioned earlier  the empty drag area is registered as a non-client caption region, so normal PointerPressed events don't fire there. The fix involves using InputNonClientPointerSource to register that area as a passthrough region, then handling the middle-click in code while keeping left-click drag behavior intact via AppWindow.

The project already uses InputNonClientPointerSource in DragZoneHelper.cs and MainPage.xaml.cs, so the pattern is established it's mostly a matter of extending it and wiring up the handler in TabBar.xaml.cs. Working through the details now and making sure window dragging doesn't break in the process. Should have something up soon.

**Comment by Yashu-svg** [comment]
Is the issue still open to work on?

**Comment by yair100** [comment]
@Yashu-svg unless @Thuram2003 is still working on it, this is open.

**Comment by Moshik520** [comment]
Hi! I'd like to work on this issue — is it still available?

My plan: handle the pointer/middle-click event on the empty area of
the tab strip and reuse the existing logic that the "+" button uses
to open a new tab, so the behavior stays consistent.

Does that sound like the right approach? This would be my first
contribution to Files, so any pointers on where to look in the
codebase would be appreciated.

**Comment by yair100** [comment]
@Moshik520 it is available, but I suggest looking at some of the previous attempts to see what didn't work.

**Comment by 0x5bfa** [comment]
suspect the system titlebar steals mouse events and dont delegate to the app itself.

**Comment by yair100** [comment]
@MazeWave this feature will be included in the next release.

**Comment by MazeWave** [comment]
> [@MazeWave](https://github.com/MazeWave) this feature will be included in the next release.

Thank you so much <3
